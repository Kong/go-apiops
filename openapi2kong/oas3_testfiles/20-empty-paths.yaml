# openapi2kong.go should not fail when .paths is empty

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/
  - url: https://server2.com/
paths: []
