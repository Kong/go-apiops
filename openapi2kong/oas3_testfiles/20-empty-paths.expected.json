{
  "_format_version": "3.0",
  "services": [
    {
      "host": "simple-api-overview.upstream",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [],
      "tags": [
        "OAS3_import",
        "OAS3file_20-empty-paths.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "id": "811c42d6-ef18-5296-a550-7dca2262b4d8",
      "name": "simple-api-overview.upstream",
      "tags": [
        "OAS3_import",
        "OAS3file_20-empty-paths.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_20-empty-paths.yaml"
          ],
          "target": "server1.com:443"
        },
        {
          "tags": [
            "OAS3_import",
            "OAS3file_20-empty-paths.yaml"
          ],
          "target": "server2.com:443"
        }
      ]
    }
  ]
}
